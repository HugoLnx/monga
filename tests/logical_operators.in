void orOperations() {
	int a, b, c;
	a = 3;
	b = 5;
	c = 2;
	if (a > c || c < b) {
		printf("a > c || c < b\n");
	} else {
		printf("! a > c || c < b\n");
	}

	if (c > a) {
		printf("c > a\n");
	} else {
		printf("! c > a\n");
	}
}

void andOperations() {
	if (3 > 2 && 7 >= 7) {
		printf("3 > 2 && 7 >= 7\n");
	}
	if (3 < 2 && 7 >= 7) {
		printf("3 > 2 && 7 >= 7\n");
	} else {
		printf("! 3 < 2 && 7 >= 7\n");
	}
	if ((3 - 2) > 5 && 5 >= 5){
		printf("(3 - 2) > 5 && 5 >= 5\n");
	} else {
		printf("! (3 - 2) > 5 && 5 >= 5\n");
	}
	if (3 > 2 && 5 <= 7) {
		printf("3 > 2 && 5 <= 7\n");
	} else {
		printf("! 3 > 2 && 5 <= 7\n");
	}
}

int main() {
	orOperations();
	andOperations();
}
