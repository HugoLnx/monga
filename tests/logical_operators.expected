a > c || c < b
! c > a
1 == 1 || 2 == 2
1 != 1 || 2 == 2
!(1 != 1 || 2 != 2)
3 > 2 && 7 >= 7
! 3 < 2 && 7 >= 7
! (3 - 2) > 5 && 5 >= 5
3 > 2 && 5 <= 7
1 == 1 && 2 == 2
!(1 != 1 && 2 == 2)
